### ISSUE TITLE
Rule Request: Switch Case RawValue

### ISSUE THREAD

**Issue by glowcap** [realm/SwiftLint#2906]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### New rule request

Switch case rawValue

1. Trigger a warning if using myEnum.rawValue to evaluate switch cases. This style forces the use of a default. Good practice could be to convert the switched value to the enum (perhaps with `guard`) prior to setting it as the switch parameter.

2. Triggers
Trigger example:

```swift
    func myFunc(myParam: String) {
      switch myParam {
      case myEnum.opt1.rawValue:
        // do something
      case myEnum.opt2.rawValue :  
        // do something else
      default:
        break // needed since myParam could be anything
      }
    }
```
 _note: spacing after `rawValue` extra space should still trigger the warning_

Not triggered if `.rawValue` isn't used

3. Only severity should be configurable

4. Rule should be Opt-in since there may be some reason the code requires a switch on `rawValue`

**Comment by glowcap** [comment]
In the meantime, I have done something in the .yml file that accomplishes the same thing (without catching `rawValue :`

```ymal
custom_rules:
  switch_raw_value:
    included: ".*\\.swift"
    name: "Switch Raw Value"
    regex: "(\\.rawValue:)"
    match_kinds:
      - identifier
    message: "Should avoid using raw values in switch cases"
    severity: warning
```
